File: base_model.py
# ItonE
class BaseAdvertising:
	'This class is the base of our other classes.'
	def __init__(self):
		print("Holy shit")
	def getClicks(self):
		return self._clicks
	def getViews(self):
		return self._views
	def incClicks(self):
		pass
	def incViews(self):
		pass
	def describeMe(self):
		return BaseAdvertising.__doc__

File: Cpp/base_model.cpp
// ItnoE
#include<bits/stdc++.h>
using namespace std;
class BaseAdvertising {
protected :
    int id = 0, clicks = 0, views = 0;
public:
    BaseAdvertising() {}
    int getClicks() {
        return (clicks);
    }
    int getViews() {
        return (views);
    }
    void incClicks() {}
    void incViews() {}
    string describeMe() {
        return ("This class is the base of our other classes.");
    }
};
